Dockerfile

```diff
@@ -0,0 +1,45 @@
+#
+# Builder image
+#
+
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.13 AS build-env
+ARG SOURCE=*
+
+ADD $SOURCE /src/
+WORKDIR /src/
+
+ENV GOFLAGS=''
+
+# Try to execute a Makefile, but if the SOURCE url is just a tar of binaries,
+# then there probably won't be one. `|| true` makes us tolerant of that case.
+RUN find . -name '*.tar.gz' -type f | xargs -rn1 tar -xzf; \
+    make || true; \
+    cp "$(find . -name 'louketo-proxy' -type f -print -quit)" /louketo-proxy
+
+#
+# Actual image
+#
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.1
```

Comment:
Why this base image (this project is not under RedHat anymore :-) )? Why static build and `FROM scratch` can't be used? It will remove dependency + minimize attack surface.

Reply:
Regarding why UBI, it'd be something I'd trust over docker-conventional busybox or alpine, and offers the same minimality as them without involving unconventional libc's.

Regarding why any parent image at all, instead of scratch, I didn't consider that. I assumed the container, as it was written for Gatekeeper, does the user setup for a reason. If that's not the case, then maybe scratch is viable.

If we were going to switch to scratch, I'd need to double check for guidance on scratch vs UBI.

Reply:
@jangaraj from your experience what is the advantage of using `scratch` instead of `UBI`?

Reply:
The security. There is no chance that any security scanner will report vulnerable/outdated packages in the image (e.g. curl, openssl, apr, ...). An administrator with image based on the `SCRATCH` will just watch for Golang security fixes. But regular image builds and container deployments will be needed with UBI image to have safe container running - for example 4 months old UBI image is vulnerable - https://catalog.redhat.com/software/containers/ubi8/ubi-minimal/5c359a62bed8bd75a2c3fba8?architecture=amd64&tag=8.1-398

Some Docker tools may have a problem with the scratch base image, e.g. CloudFoundry needs:
> The Docker image must contain an /etc/passwd file with an entry for the root user. In addition, the home directory and the shell for that root user must be present in the image file system.

I made custom `/etc/passwd` and "fake" root shell script, which is able to execute only louketo binary, so image is CloudFoundry compatible even with `scratch` base image.